pkg/controller/migmigration/annotation.go

```diff
@@ -387,7 +387,7 @@ func (t *Task) deleteAnnotations() error {
 
 // Delete Pod stage annotations and labels.
 func (t *Task) deletePodAnnotations(client k8sclient.Client) error {
-	for _, ns := range t.namespaces() {
+	for _, ns := range t.srcNamespaces() {
```

Comment:
This is run on both src and destination, so we really need srcNamespaces when we're calling it on src and destNamespaces() when calling it on dest. This might require refactoring deleteAnnotations() to explicitly grab src and dest client rather than calling getBothClients() -- and also a separate destNamespaces() function.

Reply:
Actually, thinking about it a bit more, we still want to be able to iterate over both clients instead of duplicating code. Maybe add a new fun `getBothClientsAndNamespaceList` which returns, instead of `{srcClient, destCLient}` two "client,namespaceList" pairs -- i.e. `{ {srcClient, srcNamespaces}, {destClient, destNamespaces}` -- then you can pass in the associated namespace list with the right client.